Content.Server/Administration/Managers/UsernameRuleManager.Notification.cs

```diff
@@ -0,0 +1,115 @@
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Content.Server.Database;
+
+namespace Content.Server.Administration.Managers;
+
+public sealed partial class UsernameRuleManager
+{
+    public const string UsernameRuleNotificationChannel = "username_rule_notification";
+
+    private static readonly TimeSpan UsernameRuleNotificationRateLimitTime = TimeSpan.FromSeconds(30);
+    private const int UsernameRuleNotificationRateLimitCount = 10;
+
+    private readonly object _usernameRuleNotificationRateLimitStateLock = new();
+    private TimeSpan _usernameRuleNotificationRateLimitStart;
+    private int _usernameRuleNotificationRateLimitCount;
```

Comment:
Daring approach, copy pasting this from the server ban code *without* the comments explaining why.

This rate limit doesn't make sense here, get rid of it.